crates/nu-cli/src/completions/file_completions.rs

```diff
@@ -158,8 +158,18 @@ pub fn file_path_completion(
     Vec::new()
 }
 
-pub fn matches(partial: &str, from: &str, match_algorithm: MatchAlgorithm) -> bool {
-    match_algorithm.matches_str(&from.to_ascii_lowercase(), &partial.to_ascii_lowercase())
+pub fn matches(partial: &str, from: &str, options: &CompletionOptions) -> bool {
+    // Create new values to not mutate the original ones
+    let mut haystack = from.to_string();
+    let mut needle = partial.to_string();
```

Comment:
I think it's not needed to create two new strings for case sensitive case:
```rust
if options.case_sensitive {
    options.match_algorithm.matches_str(from, partial)
} else {
    options.match_algorithm.matches_str(&from.to_ascii_lowercase(), &partial.to_ascii_lowercase())
}
```

Reply:
yeah, true